src/main/java/com/cburch/logisim/std/io/extra/Slider.java

```diff
@@ -118,46 +117,29 @@ public Object clone() {
         return null;
       }
     }
-
-    public long getCurrentValue() {
-      return this.currentvalue;
+    
+    public int getCurrentValue() {
+      final var completeValue = rightToLeft ? MAXIMUM_SLIDER_POSITION - sliderPosition : sliderPosition; 
+      return completeValue >> (MAXIMUM_NUMBER_OF_BITS - nrOfBits);
     }
-
-    public byte getCurrentX() {
-      return this.currentx;
+    
+    public int getSliderPosition() {
+      return sliderPosition;
     }
 
-    public void setCurrentBitWidth(int b) {
-      if (b != this.bitwidth) {
-        this.bitwidth = (byte) b;
-        setCurrentX();
-      }
+    public void setSliderPosition(int value) {
+      sliderPosition = (value < 0) ? 0 : (value > MAXIMUM_SLIDER_POSITION) ? MAXIMUM_SLIDER_POSITION : value; 
```

Comment:
I think you can shorten this to

    sliderPosition = Math.max(0, Math.min(value, MAXIMUM_SLIDER_POSITION));